rust/noosphere/src/ffi/tracing.rs

```diff
@@ -0,0 +1,65 @@
+use noosphere_core::tracing::{initialize_tracing, NoosphereLog};
+use safer_ffi::prelude::*;
+
+const NOOSPHERE_LOG_SILENT: u32 = 0;
+const NOOSPHERE_LOG_BASIC: u32 = 1;
+const NOOSPHERE_LOG_CHATTY: u32 = 2;
+const NOOSPHERE_LOG_INFORMED: u32 = 3;
+const NOOSPHERE_LOG_ACADEMIC: u32 = 4;
+const NOOSPHERE_LOG_TIRESOME: u32 = 5;
+const NOOSPHERE_LOG_DEAFENING: u32 = 6;
+
+#[derive_ReprC(rename = "ns_noosphere_log")]
+#[repr(u32)]
```

Comment:
Could be `u8`/`uint8_t`

Reply:
In Swift, [`CInt` is `i32`](https://developer.apple.com/documentation/swift/cint) so this makes saves from some conversions. I would use `u8` if it was similarly ergonomic to do so.

Reply:
My face as I realize the Rust side is all unsigned :facepalm:

Reply:
After more exploration, filed #345